param_sync.py

```diff
@@ -6,11 +6,42 @@ class ParamSyncRule:
     implementations of parameter synchronization rules for distributed
     training.
     """
+    def make_update_function(self, local_params):
+        """Return a function that will be called with the current value of the
+        master parameters and should update them inplace.  This
+        function must also update the values of local_params (that are
+        shared values) as a side effect.
+        """
+        try:
+            f = self.theano_update(local_params)
+            def update(master_params, f=f):
+                new_master_values = f(*master_params)
+                for p, v in zip(master_params, new_master_values):
+                    p[:] = v
+        except NotImplementedError:
+            def update(master_params, local_params=local_params,
+                       update_params=self.update_params):
+                local_param_values = [p.get_value() for p in local_params]
```

Comment:
I think the get_value() make a dependency to something that look like a shared variable. This isn't a dependency on Theano. I don't want to change this. But I just want to make this known.